archive_viewer/table_models/curve_model.py

```diff
@@ -232,13 +261,14 @@ def removeAtIndex(self, index: QModelIndex) -> None:
             An index in the row to be removed.
         """
         if isinstance(self._plot._curves[index.row()], FormulaCurveItem):
-            #Formula Curves don't have channel ddata so we should just remove it as if it were no longer valid
+            # Formula Curves don't have channel data so we should just remove it as if it were no longer valid
             self.invalidFormula(self._row_names[index.row()])
             return False
-       
-        if not index.isValid() or index.row() == (self.rowCount() - 1):
```

Comment:
We should leave this conditional as is. This prevents the user from removing the last row (an empty curve) which could get the user in a stuck state.